src/Storages/ExternalStream/Kafka/WriteBufferFromKafka.cpp

```diff
@@ -0,0 +1,80 @@
+#include "WriteBufferFromKafka.h"
+#include <cmath>
+#include <utility>
+#include "Common/logger_useful.h"
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int CANNOT_WRITE_TO_KAFKA;
+}
+
+WriteBufferFromKafka::WriteBufferFromKafka(const Poco::Logger * logger, RdKafkaTopicPtr topic_, size_t buffer_size)
+    : BufferWithOwnMemory<WriteBuffer>(buffer_size), log(logger), topic(std::move(topic_))
+{
+}
+
+void WriteBufferFromKafka::nextImpl()
+{
+    if (!offset())
+        return;
+
+    auto err = rd_kafka_produce(
+        topic.get(),
+        RD_KAFKA_PARTITION_UA,
+        RD_KAFKA_MSG_F_COPY | RD_KAFKA_MSG_F_BLOCK,
+        working_buffer.begin(),
+        offset(),
+        nullptr /* key */,
+        0 /* keylen */,
+        static_cast<void *>(this) /* opaque */);
+
+    if (unlikely(err))
+        throw Exception(
+            "Cannot write to kafka topic at offset " + std::to_string(count()) + ", error: " + rd_kafka_err2str(rd_kafka_last_error()),
+            ErrorCodes::CANNOT_WRITE_TO_KAFKA);
+
+    ++state.outstandings;
+}
+
+void WriteBufferFromKafka::finalizeImpl()
+{
+    LOG_INFO(log, "finalizeImpl is called, offset={}", offset());
```

Comment:
```c++
if (log)
    LOG_INFO(log, "finalizeImpl is called, offset={}", offset());
```

Reply:
Why `if (log)` is needed in this case?

Reply:
It appears a debug level log instead of info  since it will be too verbose.

Reply:
Ok, it is finalization, then we shall log info and let's log more information regarding topic name etc

Reply:
Removed, actually the log is not so useful.